### PR TITLE
Pass --init to docker-run to reap zombies.

### PR DESCRIPTION
**Pull request by alex3d**
Relevant docker-slave commit: jenkinsci/docker-slave@4cce983

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,21 +20,21 @@ To run a Docker container
...
 
   Linux agent:
 
-    docker run jenkins/jnlp-slave -url http://jenkins-server:port -workDir=/home/jenkins/agent <secret> <agent name>
+    docker run --init jenkins/jnlp-slave -url http://jenkins-server:port -workDir=/home/jenkins/agent <secret> <agent name>
     
   Windows agent:
   
-    docker run jenkins/jnlp-agent-windows -Url http://jenkins-server:port -WorkDir=C:/Jenkins/agent -Secret <secret> -Name <agent name>
+    docker run --init jenkins/jnlp-agent-windows -Url http://jenkins-server:port -WorkDir=C:/Jenkins/agent -Secret <secret> -Name <agent name>
```
Comment (slide) [inline README.md]: Does this work the same way on Windows?

File: README.md [hunk 2]
```diff
@@ -20,21 +20,21 @@ To run a Docker container
 
   Linux agent:
 
-    docker run jenkins/jnlp-slave -url http://jenkins-server:port <secret> <agent name>
+    docker run --init jenkins/jnlp-slave -url http://jenkins-server:port <secret> <agent name>
 
   Windows agent:
 
-    docker run jenkins/jnlp-agent:latest-windows -Url http://jenkins-server:port -Secret <secret> -Name <agent name>
+    docker run --init jenkins/jnlp-agent:latest-windows -Url http://jenkins-server:port -Secret <secret> -Name <agent name>
```
Comment (slide) [inline README.md]: Does this work the same way on Windows? Also, it might be good to note what the --init flag does.

Comment (alex3d) [reply README.md]: I don't have Windows instance to check, so I just duplicated it from jenkinsci/docker-slave.
Docker docs don't specify `--init` behavior on Windows, but from sources, it seems ignored on Windows.

File: README.md [hunk 3]
```diff
@@ -20,7 +20,7 @@ To run a Docker container
 
   Linux agent:
 
-    docker run jenkins/jnlp-slave -url http://jenkins-server:port <secret> <agent name>
+    docker run --init jenkins/jnlp-slave -url http://jenkins-server:port <secret> <agent name>
```
Comment (slide) [inline README.md]: I still think it would be good to have a Note saying why the --init flag is useful.

Comment (alex3d) [reply README.md]: Oh, I'm not good at docs wording.
> Note: `--init` is necessary for correct subprocesses handling (zombie reaping)

Is it ok?

Comment (slide) [reply README.md]: Yes, that's a good start. You could also link to the docker docs on `--init` so people can easily look further if desired.

Comment (alex3d) [reply README.md]: `docker run` docs don't provide detailed explanation either: `Run an init inside the container that forwards signals and reaps processes`.
I think it is expected that if you don't know what is "zombie reaping" you just google where do zombies come from.

Comment (alex3d) [reply README.md]: To be clear, I was not meant "you" literally, but someone who will read the docs.

**Comment by alex3d** [comment]
@slide ping

**Comment by oleg-nenashev** [comment]
I will cut the new release without it, but I am ready to cut another one shortly once this change gets accepted

**Pull request merged**